algorithm.md

```diff
@@ -0,0 +1,174 @@
+# SourceCred Algorithm
+
+SourceCred is a system for assigning 'cred' to contributors, in proportion
+to the value they've contributed to the project.
+SourceCred does this by creating a 'Contribution Graph', a [graph]
+containing all of the contributions and contributors to a project, as well as how
+they are inter-related. It then assigns cred to every node in the graph,
+with [PageRank]-esque semantics such that cred flows along edges, accumulating
+at nodes that are connected to other high-cred nodes. Intuitively,
+contributions are important if they are depended on by other important
+contributions, and users are important if they are connected to important
+contributions.
+
+The intention of this document is to rigorously describe the SourceCred algorithm,
+so that we can use it as a basis for discussing improvements and open issues with the system.
+This is a living document, and we intend to keep it up-to-date so as to describe the
+present operation of SourceCred's algorithm.
+
+Pull requests welcome!
+
+[graph]: https://en.wikipedia.org/wiki/Graph_(abstract_data_type)
+[PageRank]: https://en.wikipedia.org/wiki/PageRank
+
+
+# Contribution Graphs
+The core data structure in SourceCred is the [contribution graph]. Strictly
+speaking, it's actually a [quiver] as the edges are directed, multiple edges
+are allowed between a given pair of nodes, and loop edges are permitted.
+
+[contribution graph]: https://github.com/sourcecred/sourcecred/blob/master/src/core/graph.js
+[quiver]: https://en.wikipedia.org/wiki/Quiver_(mathematics)
+
+## Nodes
+Every contribution or contributor to a project is represented by a node in a graph.
+For example, GitHub pull requests, user accounts, comments, and repositories are all
+represented as nodes. Every node is assigned *cred* based on how it's connected to
+other nodes.
+
+It can be counter-intuitive that users get the same "kind" of cred as issues
+and pull requests, but handling things this way makes the algorithm nicely
+simple and consistent.
+
+Every node is given a unique address, which is represented by an array of
+strings. The graph does not store any metadata with nodes. Because addresses
+are unique, the addresses can be used as an index into other databases.
+
+## Edges
+Every relationship between contributors or contributors is represented by an edge in the
```

Comment:
nit "every relationship between contributors or contributors" I think you meant "contributions and contributors"

Reply:
Fixed.